File: mns-core/src/main/java/cz/fb/manaus/core/category/categorizer/SettledBetCategorizer.java
package cz.fb.manaus.core.category.categorizer;

import cz.fb.manaus.core.category.BetCoverage;
import cz.fb.manaus.core.model.SettledBet;

import java.util.Set;

public interface SettledBetCategorizer extends SimulationAware {

    default boolean isMarketSnapshotRequired() {
        return false;
    }

    Set<String> getCategories(SettledBet settledBet, BetCoverage coverage);

}

File: mns-core/src/main/java/cz/fb/manaus/core/category/categorizer/SettledBetCategorizer.kt
package cz.fb.manaus.core.category.categorizer

import cz.fb.manaus.core.category.BetCoverage
import cz.fb.manaus.core.model.SettledBet

interface SettledBetCategorizer : SimulationAware {

    val isMarketSnapshotRequired: Boolean
        get() = false

    fun getCategories(settledBet: SettledBet, coverage: BetCoverage): Set<String>

}
